From 68c5c90884e12d064924052f369313732996bfe7 Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Fri, 10 Jul 2015 17:27:55 +0530
Subject: [PATCH] USBC_PD: PD header's power_role bit set as per current role.

Currently the PD packet header's power_role bit is set always to sink.
This patch will set the bit as per current power role.

Change-Id: I1d8afc4e541516b95bcfc074c3a439c8d07d6b1e
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-13210
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
---
 drivers/usb/typec/pd/protocol.c | 6 +++++-
 include/linux/usb_typec_phy.h   | 5 +++++
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/typec/pd/protocol.c b/drivers/usb/typec/pd/protocol.c
index 3c713b4f98f7..a246e5fa767e 100644
--- a/drivers/usb/typec/pd/protocol.c
+++ b/drivers/usb/typec/pd/protocol.c
@@ -206,7 +206,11 @@ static int pd_prot_rcv_pkt_from_policy(struct pd_prot *prot, u8 msg_type,
 		pkt->header.rev_id = 1;
 	else
 		pkt->header.rev_id = 0;
-	pkt->header.pwr_role = prot->new_pwr_role;
+
+	if ((prot->new_pwr_role == PD_POWER_ROLE_PROVIDER)
+		|| (prot->new_pwr_role == PD_POWER_ROLE_CONSUMER_PROVIDER))
+		pkt->header.pwr_role = PD_PWR_ROLE_SRC;
+
 	pkt->header.msg_id = prot->tx_msg_id;
 	pkt->header.num_data_obj = len / 4;
 	memcpy((u8 *)pkt + sizeof(struct pd_pkt_header), buf, len);
diff --git a/include/linux/usb_typec_phy.h b/include/linux/usb_typec_phy.h
index 740f2753b925..4ab2682e5d6a 100644
--- a/include/linux/usb_typec_phy.h
+++ b/include/linux/usb_typec_phy.h
@@ -111,6 +111,11 @@ enum {
 };
 
 enum {
+	PD_PWR_ROLE_SINK,
+	PD_PWR_ROLE_SRC,
+};
+
+enum {
 	PD_POWER_ROLE_PROVIDER,
 	PD_POWER_ROLE_PROVIER_CONSUMER,
 	PD_POWER_ROLE_CONSUMER,
-- 
1.9.1

